cmd/junit2jira/main.go

```diff
@@ -91,9 +91,42 @@ type testIssue struct {
 	testCase j2jTestCase
 }
 
-func run(p params) error {
-	transport := http.DefaultTransport
+//region logger
+type leveledLogger struct {
+	*log.Logger
+}
+
+func (l leveledLogger) withFields(keysAndValues []interface{}) *log.Entry {
+	f := make(map[string]interface{})
+
+	for i := 0; i < len(keysAndValues)-1; i += 2 {
+		f[keysAndValues[i].(string)] = keysAndValues[i+1]
+	}
+
+	return l.WithFields(f)
+}
+
+func (l leveledLogger) Error(msg string, keysAndValues ...interface{}) {
+	l.withFields(keysAndValues).Error(msg)
+}
+
+func (l leveledLogger) Info(msg string, keysAndValues ...interface{}) {
+	l.withFields(keysAndValues).Info(msg)
+}
+func (l leveledLogger) Debug(msg string, keysAndValues ...interface{}) {
+	l.withFields(keysAndValues).Debug(msg)
+}
 
+func (l leveledLogger) Warn(msg string, keysAndValues ...interface{}) {
+	l.withFields(keysAndValues).Warn(msg)
+}
+
+//endregion
```

Comment:
Please instead move to a new `logger.go` file.

Reply:
I thought about this but so far we have everything in a single file. We can split it later in a separated PR if you like but I think having it in a single place make it easier to work with

Reply:
Adding regions sounds like it's no longer so convenient to have a single long file 😉 
I suggested a file in the same package to avoid having to export the type, but whatever...